server/pbench/bin/pbench-move-unpacked

```diff
@@ -0,0 +1,181 @@
+. $dir/pbench-base.sh
+. $dir/job_pool.sh
+###########################################################################
+
+log_init $(basename $0)
+
+# make sure only one copy is running.
+# Use 'flock -n $LOCKFILE /home/pbench/bin/pbench-move-unpacked' in the
+# crontab to ensure that only one copy is running. The script itself
+# does not use any locking.
+
+# the link source and destination for this script
+linksrc=TODO
+linkdest=TO-INDEX
+
+echo $TS
+
+# get the list of files we'll be operating on - sort them by size
+list=$TMP_PATH/pbench-unpack-tarballs.$$
+find -L $ARCHIVE/*/$linksrc -name '*.tar.xz' -printf "%s\t%p\n" 2>/dev/null | grep -v DUPLICATE | sort -n > $list
+
+trap 'rm -f $list' EXIT INT QUIT
+
+typeset -i ntb=0
+typeset -i ntotal=0
+typeset -i nerrs=0
+typeset -i ndups=0
+
+function process_tarballs {
+    result=$1
+    size=$2
+    ntotal=$ntotal+1
+
+    link=$(readlink -e $result)
+    if [ ! -f "$link" ] ;then
+        echo "$TS: $link does not exist" >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+    resultname=$(basename $result)
+    resultname=${resultname%.tar.xz}
+    hostname=$(basename $(dirname $link))
+
+    # make sure that all the relevant state directories exist
+    mk_dirs $hostname
+    # ... and a couple of other necessities.
+    if [[ $? -ne 0 ]] ;then
+        echo "$TS: Creation of $hostname processing directories failed: code $status" >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+    mkdir -p $INCOMING/$hostname
+    if [[ $? -ne 0 ]] ;then
+        echo "$TS: Creation of $INCOMING/$hostname failed: code $status" >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+    
+    # XXXX - for now, if it's a duplicate name, just punt and avoid producing the error - the full
+    # solution will involve renaming the unpacked directory appropriately.
+    if [ ${resultname%%.*} == "DUPLICATE__NAME" ] ;then
+        ndups=$ndups+1
+        continue
+    fi
+
+    incoming=$INCOMING/$hostname/$resultname
+    mkdir -p $incoming
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: mkdir -p $incoming failed: code $status" >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    let start_time=$(date +%s)
+    PROG="pbench-unpack-tarballs"
+    pushd $TMP_PATH/$PROG/$hostname > /dev/null 2>&1
+    popd > /dev/null 2>&1
+
+    # copy the tarball contents to INCOMING - remove the link on failure
+    cp -R $TMP_PATH/$PROG/$hostname/$resultname $incoming.copy
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: Cannot copy $TMP_PATH/$PROG/$hostname/$resultname to $incoming.copy: code $status" >&4
+        rm $RESULTS/$hostname/$prefix$resultname
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # remove the symlink from incoming
+    rm $incoming
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: Cannot remove $incoming: code $status" >&4
+	rm $RESULTS/$hostname/$prefix$resultname
+        nerrs=$nerrs+1
+        continue
+    fi
```

Comment:
I'm not convinced that this is the right way to go about it: the copy of the directory has succeeded, so even though the $incoming symlink (btw, you should check that it *is* a symlink - if it is a directory, then something is screwy) cannot be deleted, we can change the $RESULTS/... symlink to point to the copy. To be sure, cleanup will be required, but it seems to me that there is no need to remove the $RESULTS symlink in this case and we should probably go on, rather than abandoning this tarball.